dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php

```diff
@@ -0,0 +1,469 @@
+            message: "Lorem Ipsum"
+        }          
+          recipients: [
+              {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Name of sender.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testSendWithoutSenderEmail()
+    {
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1
+          sender: {
+            name: "Name"
+            email: ""
+            message: "Lorem Ipsum"
+        }          
+          recipients: [
+              {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Email of sender.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testSendWithoutSenderMessage()
+    {
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1	
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: ""
+        }          
+          recipients: [
+              {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Message.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     * @magentoApiDataFixture Magento/SendFriend/Fixtures/sendfriend_configuration.php
```

Comment:
Generally, we should use the approach described in `https://github.com/magento/graphql-ce/issues/167`
Pls, link this temporal solution with the https://github.com/magento/graphql-ce/issues/167

Reply:
As far as I understand I can skip this comment. Just leave todo.